### DISCUSSION TITLE
Tree / timeline representation of the cooking process

### DISCUSSION THREAD

**Discussion by tobei** [cooklang/spec#48] [Ideas]
When i'm reading a recipe with multiple preparations (e.g. main product with marinade, sauce, veggies...) I usually like to represent myself the recipe as an inverted tree starting with the leafs (my ingredients) and merging the branches together into the final products (mardinade, meat, sauce, veggies) then into bigger product until reaching the root (final product). 

I make the branches collapse in the order i need to add/mix items in the order i should add items in, and even add on the branches the optional wait time before doing so or further instruction like ("blend with a mixer until consistancy reach...", "cook until ...", "rest in the fridge for 20minutes"). I find that pretty easy to visualise what steps i'm in, and whats need to be done.

While some of the instructions in the current format allows for timer and instructions, I think the step by step format being for the whole dish doesn't make it easy to approach and separate the different parts of a dish. Additionally some parts of a dish (for example the sauce) could later be extracted in their own recipe and contribute (add) to the ingredients of another dish. A tree based vision of cooking (while keeping it readable first !) allows for easy graphical representation during cooking, but also easy refactoring (separating, merging or making new recipe from smaller ones). Perhaps syntax should allow for that "depends on [+ time/actions]" relation between the steps so that a visual tree/timeline can be created and part of recipe externalized/internalized (supporting merge is also a must)

Maybe my idea is a big strange... but i always make that tree as it also helps me prepare my ingredients and make sure i didn't forget to add anything ^^

**Comment by dubadub** [comment]
Linking to a related discussion in issues https://github.com/cooklang/spec/issues/36.

**Reply by aspiers** [reply]
@tobei Your idea isn't at all strange, it sounds exactly what I proposed in #62 (recently moved to discussions, previously at the link above - thanks to @dubadub for sharing that).
